src/otx/algo/classification/metrics/hlabel_accuracy.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for defining hierarchical label accuracy metric."""
+
+from __future__ import annotations
+
+import torch
+from torchmetrics import Metric
+from torchmetrics.classification import Accuracy, MultilabelAccuracy
+
+
+class HLabelAccuracy(Metric):
+    """Custom accuracy metric for h-label classification.
+
+    Args:
+        num_multiclass_heads (int): Number of multi-class heads.
+        num_multilabel_classes (int): Number of multi-label classes.
+        threshold_multilabel (float): Predictions with scores under the thresholds
+                                        are considered as negative. Defaults to 0.5.
+        head_idx_to_logits_range (dict[str, tuple[int, int]]): The range of logits which represents
+                                                                the number of classes for each heads.
+    """
+
+    def __init__(
+        self,
+        num_multiclass_heads: int,
+        num_multilabel_classes: int,
+        threshold_multilabel: float = 0.5,
+        head_idx_to_logits_range: dict[str, tuple[int, int]] | None = None,
+    ):
+        super().__init__()
+
+        self.num_multiclass_heads = num_multiclass_heads
+        self.num_multilabel_classes = num_multilabel_classes
+        self.threshold_multilabel = threshold_multilabel
+        self.head_idx_to_logits_range = head_idx_to_logits_range
+
+        # Multiclass classification accuracy will be defined later
+        self.multiclass_head_accuracy: list[Accuracy] | None = None
+
+        # Multilabel classification accuracy metrics
+        if num_multilabel_classes > 0:
+            # self.multilabel_accuracy = MultilabelAveragePrecision(
+            #     num_labels=self.num_multilabel_classes, average='macro'
+            # )
```

Comment:
Please check again that there are commented lines

Reply:
Thanks, removed. https://github.com/openvinotoolkit/training_extensions/pull/2761/commits/fdc033ec6b1fdc6615705ddd75af960a49af9f1d